### DISCUSSION TITLE
Where to get new Session on login (sessionToken) before it's send to the client?

### DISCUSSION THREAD

**Discussion by devklepacki** [nextauthjs/next-auth#6406] [Help]
### Question 💬

TL;DR: **Is there an accessible place to hook up when a Session is already created in the database (when a User signs in), but the response is not yet sent to a client?**

I'm using `database` strategy with Prisma. I have an app that allows for having multiple Schools per user. A user may have a different role in each School.

I already have an entire system around this in place, but I want to allow logging into a different School per device, which basically translates into a separate School per Session. So right now when the User switches Schools in UI I grab their's `sessionToken` cookie on the backend, search for it in the `sessions` table, and add/update `schoolId` and `role` fields (for this session, in `sessions` table).

But how do I do this when a User logs in? There's no cookie with `sessionToken` yet.

**I cannot find any place to hook up when a Session is already created in the database when a User signs in, but the response is not yet sent to a client.**

I really don't want to first sign them in, then do another hop to the backend to update `session` table, and only then redirect them to the actual app. I even desperately tried to:
```tsx
// /api/auth/[...nextauth].ts
(...)
const nextAuthResponse = await NextAuth(req, res, authOptions)
console.log(nextAuthResponse) // undefined
const nothingIsHere = res.getHeader('Set-Cookie') // undefined
return nextAuthResponse
```
but with no luck.

Or maybe this entire approach is flawed and should be approached differently from the beginning?

### How to reproduce ☕️

---

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR
